src/wrapped_radix_engine_toolkit/low_level/transaction_manifest/execution_summary/transaction_manifest_execution_summary.rs

```diff
@@ -0,0 +1,921 @@
+
+impl TransactionManifest {
+    pub fn execution_summary(
+        &self,
+        encoded_receipt: BagOfBytes,
+    ) -> Result<ExecutionSummary> {
+        let receipt: TransactionReceipt = encoded_receipt.try_into()?;
+        self.execution_summary_with_receipt(receipt)
+    }
+
+    fn execution_summary_with_receipt(
+        &self,
+        receipt: TransactionReceipt,
+    ) -> Result<ExecutionSummary> {
+        let ret_execution_summary =
+            RET_execution_summary(&self.scrypto_manifest(), &receipt.decoded)
+                .map_err(|e| {
+                error!(
+                    "Failed to get execution summary from RET, error: {:?}",
+                    e
+                );
+                CommonError::FailedToGetRetExecutionSummaryFromManifest
+            })?;
+
+        Ok(ExecutionSummary::from((
+            ret_execution_summary,
+            self.network_id(),
+        )))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use radix_engine::transaction::{
+        AbortReason, AbortResult, TransactionResult,
+    };
+
+    use super::*;
+
+    #[allow(clippy::upper_case_acronyms)]
+    type SUT = ExecutionSummary;
+
+    #[test]
+    fn invalid_receipt() {
+        assert_eq!(
+            TransactionManifest::sample()
+                .execution_summary(BagOfBytes::from_hex("dead").unwrap()),
+            Err(CommonError::FailedToDecodeEncodedReceipt)
+        );
+    }
+
+    #[test]
+    fn failure_if_receipt_result_is_abort() {
+        let encoded_receipt_hex = include_str!("create_pool.dat");
+        let wrong_receipt_raw =
+            BagOfBytes::from_hex(encoded_receipt_hex).unwrap();
+        let mut wrong_receipt: TransactionReceipt =
+            wrong_receipt_raw.try_into().unwrap();
+        wrong_receipt.decoded.result = TransactionResult::Abort(AbortResult {
+            reason: AbortReason::ConfiguredAbortTriggeredOnFeeLoanRepayment,
+        });
+        assert_eq!(
+            TransactionManifest::sample()
+                .execution_summary_with_receipt(wrong_receipt),
+            Err(CommonError::FailedToGetRetExecutionSummaryFromManifest)
+        );
+    }
+
+    impl Default for FeeLocks {
+        fn default() -> Self {
+            Self::new(0, 0)
+        }
+    }
+
+    #[test]
+    fn transfer_1to2_multiple_nf_and_f_tokens() {
+        let encoded_receipt_hex =
+            include_str!("transfer_1to2_multiple_nf_and_f_tokens.dat");
+        let instructions_string =
+            include_str!("transfer_1to2_multiple_nf_and_f_tokens.rtm");
+
+        let transaction_manifest = TransactionManifest::new(
+            instructions_string,
+            NetworkID::Stokenet,
+            Blobs::default(),
+        )
+        .unwrap();
+
+        let sut = transaction_manifest
+            .execution_summary(
+                BagOfBytes::from_hex(encoded_receipt_hex).unwrap(),
+            )
+            .unwrap();
+
+        pretty_assertions::assert_eq!(
```

Comment:
I have to convert the rest of the assertions in this mod tests to this, will be done tomorrow morning with that.